src/main/java/info/tehnut/soulshards/item/ItemSoulShard.java

```
@@ -35,15 +35,23 @@
 
 public class ItemSoulShard extends Item implements ISoulShard {
 
-    private static final MethodHandle GET_SPAWNER_ENTITY;
+    private static MethodHandle GET_SPAWNER_ENTITY;
 
     static {
         try {
             Method _getEntityName = MobSpawnerLogic.class.getDeclaredMethod("method_8281");
             _getEntityName.setAccessible(true);
             GET_SPAWNER_ENTITY = MethodHandles.lookup().in(MobSpawnerLogic.class).unreflect(_getEntityName);
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            try {
+                Method _getEntityName = MobSpawnerLogic.class.getDeclaredMethod("getEntityId");
+                _getEntityName.setAccessible(true);
+                GET_SPAWNER_ENTITY = MethodHandles.lookup().in(MobSpawnerLogic.class).unreflect(_getEntityName);
+            }
+            catch (Exception f)
+            {
+                f.printStackTrace();
+            }
```

You should really be using Fabric Loader's Mappings API here. Yarn names change fast, and this will not work with a new name. 
Also, this exception should not be silently swallowed. Not resolving the method is a significant error condition that will cause a NPE down the line.

---

Changed from reflection to a mixin invoker as we discussed in discord.